app.go

```diff
@@ -109,15 +109,21 @@ func (app *App) AddHandler(spec Spec) error {
 		return fmt.Errorf("the spec does not provide a handler function: %v", spec)
 	}
 
-	// register the handler in the router
-	route := app.router.HandleFunc(spec.Path, handler).Methods(spec.Methods...)
-	if len(spec.Headers) != 0 {
-		route.Headers(spec.Headers...)
+	// backward compatibility
+	if spec.Path != "" {
```

Comment:
Sounds good. Done. Ideally these two should be mutually exclusive but it's good for now.